src/object/srv_obj.c

```diff
@@ -983,7 +960,7 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	if (orw->orw_shard_tgts.ca_arrays == NULL) {
```

Comment:
I think we need further discussion on single replica case.  If you insist to have it, I can do that by some other in-memory structure for single replica resend, but not by dtx, which I thought confuse thing.  Anyway, we do not have any requirement or tests for that yet.

Reply:
I have made some resent RPC test cases, but not commit yet because of some other rebuild tests against DTX. On the other hand, I will work on conditional update very soon, that also requires to handle resent RPC for single replica case. So we need handle resent RPC for single replica case. I agree that DTX-based resent handling is not perfect, but we have to keep it until we have better solution.

Reply:
I can work on some in-memory resend check(without dtx) first. As to how to store it in pmem, and how to shrink it,  I think we need further discussion.

Reply:
About the resend discussion is out of this patch scope. But before we get better solution, I prefer to keep original implementation, the current implementation (made by your former patch (#443) breaks the resent-RPC handling.